tests/Sort.test.ts

```diff
@@ -222,45 +287,179 @@ describe('Sort', () => {
         expect(
             Sort.by(
                 {
-                    sorting: [{ property: 'description', reverse: false }],
+                    sorting: [
+                        {
+                            property: 'description',
+                            reverse: false,
+                            propertyInstance: 0,
+                        },
+                    ],
                 },
                 [two, one, five, four, three],
             ),
         ).toEqual(expectedOrder);
     });
+});
 
-    it('sorts correctly by tag, done', () => {
-        const one = fromLine({
-            line: '- [ ] a #someday #home 📅 1970-01-01 ✅ 1971-01-01',
-            path: '',
-        });
-        const two = fromLine({
-            line: '- [ ] #task b #someday #home 📅 1970-01-02 ✅ 1971-01-02',
-            path: '',
-        });
-        const three = fromLine({
-            line: '- [ ] c #next #home 📅 1970-01-02 ✅ 1971-01-01',
-            path: '',
-        });
-        const four = fromLine({
-            line: '- [ ] d #urgent #home 📅 1970-01-02 ✅ 1971-01-03',
-            path: '',
-        });
-        const five = fromLine({
-            line: '- [ ] e 📅 1970-01-02 ✅ 1971-01-03',
-            path: '',
-        });
-        const expectedOrder = [three, one, two, four, five];
+describe('Sort by tags', () => {
+    it('should sort correctly by tag defaulting to first with no global filter', () => {
+        const t1 = fromLine({ line: '- [ ] a #aaa #jjj' });
+        const t2 = fromLine({ line: '- [ ] a #ggg #ccc' });
+        const t3 = fromLine({ line: '- [ ] a #bbb #iii' });
+        const t4 = fromLine({ line: '- [ ] a #fff #aaa' });
+        const t5 = fromLine({ line: '- [ ] a #ccc #bbb' });
+        const t6 = fromLine({ line: '- [ ] a #eee #fff' });
+        const t7 = fromLine({ line: '- [ ] a #ddd #ggg' });
+        const t8 = fromLine({ line: '- [ ] a #hhh #eee' });
+        const t9 = fromLine({ line: '- [ ] a #iii #ddd' });
+        const t10 = fromLine({ line: '- [ ] a #jjj #hhh' });
+        const expectedOrder = [t1, t3, t5, t7, t6, t4, t2, t8, t9, t10];
+        expect(
```

Comment:
I like being able to reason about the expected behaviour in the tests...
So I spent several minutes trying to confirm, mentally, that the order of values in `expectedOrder` really was correct.

In the end, I manually sorted the lines of code by first tag, so that the code looked like this:

```typescript
        const t1 = fromLine({ line: '- [ ] a #aaa #jjj' });
        const t3 = fromLine({ line: '- [ ] a #bbb #iii' });
        const t5 = fromLine({ line: '- [ ] a #ccc #bbb' });
        const t7 = fromLine({ line: '- [ ] a #ddd #ggg' });
        const t6 = fromLine({ line: '- [ ] a #eee #fff' });
        const t4 = fromLine({ line: '- [ ] a #fff #aaa' });
        const t2 = fromLine({ line: '- [ ] a #ggg #ccc' });
        const t8 = fromLine({ line: '- [ ] a #hhh #eee' });
        const t9 = fromLine({ line: '- [ ] a #iii #ddd' });
        const t10 = fromLine({ line: '- [ ] a #jjj #hhh' });
        const expectedOrder = [t1, t3, t5, t7, t6, t4, t2, t8, t9, t10];
```

I could then glance at the order of variables in `expectedOrder` and see that indeed, the order was correct.

It left me wondering how the values in the 2nd tag where chosen.
It also left me wondering what the value of adding 10 data points was, and whether 3 or 4 would have been enough, and have made the test easier to understand.

Reply:
We already have similar tests for sorting. I had the same issue with the original tests back then and I changed them. See for example `Sort.test.ts` lines 23 ff.

To be consistent, we should do it here the same way we do it there:
1. Declare the variables in the correct order with the correct names.
2. Add them to the test case in mixed up order.

So something like:

```ts
        const t1 = fromLine({ line: '- [ ] a #aaa #jjj' });
        const t2 = fromLine({ line: '- [ ] a #bbb #iii' });
        const t3 = fromLine({ line: '- [ ] a #ccc #bbb' });
        const t4 = fromLine({ line: '- [ ] a #ddd #ggg' });
        const t5 = fromLine({ line: '- [ ] a #eee #fff' });
        // ... and so on
        const expectedOrder = [t1, t2, t3, t4, t5 /* ... and so on */];
```

And then something like:

```ts
            Sort.by(
                {
                    sorting: [
                        {
                            property: 'tag',
                            reverse: false,
                            propertyInstance: 0,
                        },
                    ],
                },
                [t5, t3, t2, t4, t1 /* ... and so on */], // Mix them here!
            ),
```

Reply:
Makes sense, I will reorder them to follow that convention.